### PR TITLE
Refactor + Vertex

### PR DESCRIPTION
**Pull request by KarolDenst**
Added vertex struct.
Fixed formatting.
Added scene class.
Not everything is done but i want to create this PR to avoid future conflicts since it changes almost every file in the solution.

### CODE REVIEW

File: Hyper/HUD/HUDElements/FPSCounter.cs [hunk 1]
```diff
@@ -4,35 +4,34 @@
...
 namespace Hyper.HUD.HUDElements
 {
-    internal class FPSCounter : HUDElement
+    internal class FpsCounter : HudElement
     {
-        internal const float DefaultSize = 0.02f;
+        public const float DefaultSize = 0.02f;
 
-        internal static Vector2 DefaultPosition = new Vector2(0.64f, 0.48f);
+        public static Vector2 DefaultPosition = new(0.64f, 0.48f);
```
Comment (AleksyBalazinski) [inline Hyper/HUD/HUDElements/FPSCounter.cs:11]: can it be readonly?

File: Hyper/HUD/HUDManager.cs [hunk 2]
```diff
@@ -5,25 +5,28 @@
 
 namespace Hyper.HUD
 {
-    internal class HUDManager : Commandable
+    internal class HudManager : Commandable
     {
-        private Shader _shader;
+        public float AspectRatio;
```
Comment (AleksyBalazinski) [inline Hyper/HUD/HUDManager.cs:10]: maybe property?

File: Hyper/Meshes/Chunk.cs [hunk 3]
```diff
@@ -11,28 +12,16 @@ internal class Chunk : Mesh
 
         public new Vector3i Position;
 
-        public Vector3i Middle => Position + Vector3i.One * Size / 2;
+        private readonly float[,,] _voxels;
 
-        private float[,,] _voxels;
+        private static readonly Logger Logger = LogManager.GetCurrentClassLogger();
```
Comment (AleksyBalazinski) [inline Hyper/Meshes/Chunk.cs:17]: start var name with _

File: Hyper/Meshes/Chunk.cs [hunk 4]
```diff
@@ -94,39 +84,50 @@ public bool Build(Vector3 location, float deltaTime, int radius = 5)
...
+            Vertex[] vertices = renderer.GetMesh();
+            NumberOfVertices = vertices.Length;
 
             GL.BindVertexArray(VaoId);
             GL.DeleteBuffer(VboId);
             VboId = GL.GenBuffer();
 
             GL.BindBuffer(BufferTarget.ArrayBuffer, VboId);
-            GL.BufferData(BufferTarget.ArrayBuffer, vertices.Length * sizeof(float), vertices, BufferUsageHint.StaticDraw);
+            GL.BufferData(BufferTarget.ArrayBuffer, vertices.Length * Marshal.SizeOf<Vertex>(), vertices, BufferUsageHint.StaticDraw);
```
Comment (AleksyBalazinski) [inline Hyper/Meshes/Chunk.cs:109]: again not sure why Marshal.SizeOf()

File: Hyper/Meshes/Mesh.cs [hunk 5]
```diff
@@ -33,19 +32,19 @@ public virtual void Render(Shader shader, float scale, Vector3 cameraPosition)
...
-        private void CreateVertexArrayObject(float[] vertices)
+        private void CreateVertexArrayObject(Vertex[] vertices)
         {
             int vaoId = GL.GenVertexArray();
             GL.BindVertexArray(vaoId);
 
             int vboId = GL.GenBuffer();
             GL.BindBuffer(BufferTarget.ArrayBuffer, vboId);
-            GL.BufferData(BufferTarget.ArrayBuffer, vertices.Length * sizeof(float), vertices, BufferUsageHint.StaticDraw);
+            GL.BufferData(BufferTarget.ArrayBuffer, vertices.Length * Marshal.SizeOf<Vertex>(), vertices, BufferUsageHint.StaticDraw);
```
Comment (AleksyBalazinski) [inline Hyper/Meshes/Mesh.cs:42]: as above

File: Hyper/Meshes/Vertex.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
+﻿using System.Runtime.InteropServices;
+
+namespace Hyper.Meshes;
+
+[StructLayout(LayoutKind.Explicit)]
+internal struct Vertex
```
Comment (AleksyBalazinski) [inline Hyper/Meshes/Vertex.cs:6]: We can rename it to Vertex3d to be consistent

File: Hyper/Texture.cs [hunk 7]
```diff
@@ -5,11 +5,11 @@
 
 namespace Hyper
 {
-    public class Texture
+    internal class Texture
     {
-        private const int _fontSize = 50;
+        private const int FontSize = 50;
```
Comment (AleksyBalazinski) [inline Hyper/Texture.cs:10]: start with _

File: Hyper/Window.cs [hunk 8]
```diff
@@ -116,9 +73,9 @@ protected override void OnUpdateFrame(FrameEventArgs e)
                 Close();
             }
 
-            UpdateCamera(input, time);
+            _scene.UpdateCamera(input, time, new Vector2(MouseState.X, MouseState.Y));
```
Comment (AleksyBalazinski) [inline Hyper/Window.cs:76]: I feel like we're exposing too much details about the scene. Maybe something like `_scene.Update(<all the necessary stuff>)` but this is ugly too. If I think of a better solution I'll push it in my commit

File: Hyper/Window.cs [hunk 9]
```diff
@@ -10,31 +8,13 @@
 
 namespace Hyper
 {
-    public class Window : GameWindow
+    internal class Window : GameWindow
     {
-        private static Logger _logger = LogManager.GetCurrentClassLogger();
+        private static readonly Logger Logger = LogManager.GetCurrentClassLogger();
```
Comment (AleksyBalazinski) [inline Hyper/Window.cs:13]: _

**Review by AleksyBalazinski — CHANGES_REQUESTED** [review]
Added a few comments

File: Hyper/HUD/HUDElements/Crosshair.cs [hunk 10]
```diff
@@ -3,28 +3,27 @@
 
 namespace Hyper.HUD.HUDElements
 {
-    internal class Crosshair : HUDElement
+    internal class Crosshair : HudElement
     {
-        internal const float DefaultSize = 0.02f;
+        public const float DefaultSize = 0.02f;
```
[the remark this answers is not in the document]
Comment (KarolDenst) [reply Hyper/HUD/HUDElements/Crosshair.cs:8]: I don't think it applies to constants. Will correct in other places

File: Hyper/Meshes/Vertex.cs (hunk 6 above)
Comment (KarolDenst) [reply Hyper/Meshes/Vertex.cs:6]: how about changing Vertex2d to HUDVertex? Also I think we might want to move HUD to another projectsince it doesnt interact with anything in this one. But let's leave that for a later PR.

Comment (KarolDenst) [reply Hyper/Meshes/Vertex.cs:6]: Then both could be called Vertex

File: Hyper/Window.cs (hunk 8 above)
Comment (KarolDenst) [reply Hyper/Window.cs:76]: Maybe add structs like MouseArguments, TimeArguments and so on and them pass them to the Update() method?

**Review by AleksyBalazinski — APPROVED** [review]

**Pull request merged**